### ISSUE TITLE
Whitespace at  the end of clox' snippets are breaking macros

### ISSUE THREAD

**Issue by brandizzi** [munificent/craftinginterpreters#276]
The clox examples have a lot of extra whitespaces at then end of the lines. This causes problem when we are defining multiline macros. For example, when I copy and paste the GROW_CAPACITY() macro, I get this error:

```
memory.h:4:33: warning: backslash and newline separated by space
 #define GROW_CAPACITY(capacity) \        
                                 ^
```

Since there was no such dangling whitespaces in jlox snippets, I wonder: could it be removed here? (If possible, I have to confess, I would love to have all dangling whitespace removed, not only the ones affecting macros: they give me shivers :grimacing:)

**Comment by munificent** [comment]
Ooh, gnarly.

I added the trailing spaces to fix this tricky issue: #252. I'll see if I can find a more elegant way to handle that, though CSS (as usual) doesn't make it easy.

**Comment by jptemplin** [comment]
I bump this. For those of us who aren't C preprocessor fanatics, the compiler errors you get because of the trailing spaces are initially baffling. I only realized what the problem was by downloading your complete source from github and comparing the macros I was copying/pasting vs those in your working source code files.
